source/DSCResources/DSC_CMAssetIntelligencePoint/DSC_CMAssetIntelligencePoint.psm1

```diff
@@ -0,0 +1,446 @@
+$script:dscResourceCommonPath = Join-Path -Path $PSScriptRoot -ChildPath '..\..\Modules\DscResource.Common'
+$script:configMgrResourcehelper = Join-Path -Path $PSScriptRoot -ChildPath '..\..\Modules\ConfigMgrCBDsc.ResourceHelper'
+
+Import-Module -Name $script:dscResourceCommonPath
+Import-Module -Name $script:configMgrResourcehelper
+
+$script:localizedData = Get-LocalizedData -DefaultUICulture 'en-US'
+
+ <#
+    .SYNOPSIS
+        This will return a hashtable of results.
+
+    .PARAMETER SiteCode
+        Specifies a site code for the Configuration Manager site that manages the system role for the asset intelligence point.
+
+    .PARAMETER SiteServerName
+        Specifies the Site Server to install or configure the role on.
+
+    .Notes
+        This role must only be installed on top-level site of the hierarchy.
+
+#>
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteServerName
+    )
+
+    Write-Verbose -Message $script:localizedData.RetrieveSettingValue
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+
+    $apProps = Get-CMAssetIntelligenceProxy
+
+    if ($apProps)
+    {
+        $cert         = $apProps.ProxyCertPath
+        $apEnabled    = $apProps.ProxyEnabled
+        $syncEnabled  = $apProps.PeriodicCatalogUpdateEnabled
+        $syncSchedule = $apProps.PeriodicCatalogUpdateSchedule
+        $status       = 'Present'
+
+        $convertCimParam = @{
+            ScheduleString = $syncSchedule
+            CimClassName   = 'DSC_CMAssetIntelligenceSynchronizationSchedule'
+        }
+        $sSchedule = ConvertTo-CimCMScheduleString @convertCimParam
+    }
+    else
+    {
+        $status = 'Absent'
+    }
+
+    return @{
+        SiteServerName        = $SiteServerName
```

Comment:
As discussed, can we pull what server the role is actually installed on?

Reply:
Changed to pull server name that the role is installed on. Updated pester to reflect change.